### PR TITLE
Window: general refactor and move to vulkan-hpp

### PR DESCRIPTION
**Pull request by MoonDweller29**

### CODE REVIEW

File: src/magma/vk/Window.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+#include <iostream>
+#include "magma/vk/vulkan_common.h"
+
+void Window::framebufferResizeCallback(GLFWwindow* window, int width, int height) {
+//    bool volatile loop = true;
+    Window *app_window = reinterpret_cast<Window *>(glfwGetWindowUserPointer(window));
+    app_window->_wasResized = true;
+//    while (loop);
+//    std::cout << "RESIZED\n";
```
Comment (Nastormo) [inline src/magma/vk/Window.cpp:11]: Need delete

File: src/magma/vk/Window.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
+#include "magma/vk/Window.h"
+
+#include <iostream>
+#include "magma/vk/vulkan_common.h"
+
+void Window::framebufferResizeCallback(GLFWwindow* window, int width, int height) {
+//    bool volatile loop = true;
```
Comment (Nastormo) [inline src/magma/vk/Window.cpp:7]: Need delete

File: include/magma/vk/Window.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    GLFWwindow * getGLFWp()         { return _window;         }
+    VkSurfaceKHR getSurface() const { return _surface;        }
+    bool         wasResized() const { return _wasResized;     }
+    Keyboard *   getKeyboard()      { return _keyboard.get(); }
+    Mouse *      getMouse()         { return _mouse.get();    }
+
+private:
+    uint32_t                   _width, _height;
+    bool                       _wasResized = false;
```
Comment (Nastormo) [inline include/magma/vk/Window.h:32]: Move to constructor

**Review by Nastormo — DISMISSED** [review]

File: src/magma/vk/Window.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    _keyboard = std::make_unique<Keyboard>(_window);
+    _mouse = std::make_unique<Mouse>(this);
+
+    _surface = createSurface(_instance, _window);
+}
+
+void Window::initContext() {
+    glfwInit();
+}
+void Window::closeContext() {
```
Comment (Nastormo) [inline src/magma/vk/Window.cpp:46]: add \n

**Review by Nastormo — CHANGES_REQUESTED** [review]

File: src/magma/vk/Window.cpp (hunk 1 above)
Comment (MoonDweller29) [reply src/magma/vk/Window.cpp]: done

File: src/magma/vk/Window.cpp (hunk 2 above)
Comment (MoonDweller29) [reply src/magma/vk/Window.cpp]: done

File: include/magma/vk/Window.h (hunk 3 above)
Comment (MoonDweller29) [reply include/magma/vk/Window.h]: done

**Review by Nastormo — APPROVED** [review]

**Pull request merged**